.github/workflows/install.yml

```diff
@@ -35,18 +35,98 @@ jobs:
           key: ${{ runner.os }}-${{ hashFiles('opam.locked') }}
 
       - name: Install Required Brew Packages for MacOS
-        run: brew install pkg-config automake
+        run: brew install pkg-config automake jq
         if: runner.os == 'macOS'
 
-      # ensure infer isn't installed in this switch, then deal with dependencies
-      - run: opam remove infer || true
+      - name: Install Required Apt Packages for Ubuntu
+        run: sudo apt-get install libmpfr-dev libsqlite3-dev
+        if: runner.os == 'Linux'
 
-      - run: opam update --upgrade
+      - name: Fetch clang Release
+        run: |
+          REPO="facebook/facebook-clang-plugins"
+          URL="https://api.github.com/repos/$REPO"
 
-      - run: opam pin add --no-action infer .
+          CLANGFILE="clang-install-${{ matrix.os }}.tar.gz"
 
-      - run: opam depext --update infer;
+          RELEASE_INFO="$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "$URL/releases/latest")"
+          if ! ASSET_ID="$(jq ".assets | map(select(.name == \"$CLANGFILE\"))[0].id" <<< "$RELEASE_INFO")"; then
+              echo "$RELEASE_INFO"
+              exit 1
+          fi
 
-      - run: opam install --deps-only infer
+          if [[ -z "$ASSET_ID" || "$ASSET_ID" == "null" ]]; then
+              echo "Could not find asset ID" >&2
+              exit 1
+          fi
 
-      - run: opam install infer
+          curl -J -L "$URL/releases/assets/$ASSET_ID" \
+              -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+              -H "Accept: application/octet-stream" \
+              -o "$CLANGFILE"
+
+      - run: tar -xzf clang-install-${{ matrix.os }}.tar.gz -C facebook-clang-plugins/clang/
+
+      - run: ./facebook-clang-plugins/clang/setup.sh --only-record-install
+
+      - run: ./build-infer.sh -y all
```

Comment:
Replacing the `opam install infer` by this line is a major improvement IMO. Opam horribly truncates the build log, and it is almost impossible to understand why the build failed. This is much better!